utils/crypto/bls/signer.go

```diff
@@ -4,7 +4,7 @@
 package bls
 
 type Signer interface {
-	PublicKey() *PublicKey
-	Sign(msg []byte) *Signature
-	SignProofOfPossession(msg []byte) *Signature
+	PublicKey() (*PublicKey, error)
```

Comment:
Why does this need an error? Can't you the implementation cache the public key when the signer is initialized since it's always constant?